nautobot/dcim/tests/test_views.py

```diff
@@ -2986,3 +2990,48 @@ def setUpTestData(cls):
             "long_term_support": False,
             "pre_release": True,
         }
+
+
+class ControllerTestCase(ViewTestCases.PrimaryObjectViewTestCase):
+    model = Controller
+    filterset = ControllerFilterSet
+
+    @classmethod
+    def setUpTestData(cls):
+        statuses = Status.objects.get_for_model(Controller)
+        roles = Role.objects.get_for_model(Controller)
+        platforms = Platform.objects.all()
+        locations = Location.objects.all()
+
+        cls.form_data = {
+            "name": "Controller 1",
+            "platform": platforms[0].pk,
+            "status": statuses[0].pk,
+            "role": roles[0].pk,
+            "location": locations[0].pk,
+        }
+
+        cls.bulk_edit_data = {
+            "platform": platforms[0].pk,
+            "status": statuses[0].pk,
+            "role": roles[0].pk,
+        }
+
+
+class ControllerDeviceGroupTestCase(ViewTestCases.PrimaryObjectViewTestCase):
+    model = ControllerDeviceGroup
+    filterset = ControllerDeviceGroupFilterSet
+
+    @classmethod
+    def setUpTestData(cls):
+        controllers = Controller.objects.all()
+
+        cls.form_data = {
+            "name": "Controller Device Group 10",
+            "controller": controllers[0].pk,
+            "weight": 100,
```

Comment:
Similarly, should the Device view test `bulk_edit_data` include a `controller_device_group` value now?

Reply:
> I would expect/hope that creating/editing a ControllerDeviceGroup would allow me to select the set of devices that belong to that group (as an alternative to editing each device individually to assign it to the group). Is that possible?

Added as a todo to the base PR.

Reply:
> Similarly, should the Device view test `bulk_edit_data` include a `controller_device_group` value now?

Added setting `controller_device_group` to the `bulk_edit_data` test.